collector/container.go

```diff
@@ -327,6 +281,66 @@ func (c *ContainerMetricsCollector) collect(ch chan<- prometheus.Metric) (*prome
 		)
 	}
 
+	// Collect network stats by querying directly from HNS
+	hnsEndpoints, err := hcsshim.HNSListEndpointRequest()
+	if err != nil {
+		log.Warn("Failed to collect network stats for containers because HNS list endpoints call failed")
+		return nil, nil
+	}
+
+	if len(hnsEndpoints) == 0 {
+		log.Info("No network stats for containers to collect")
+		return nil, nil
+	}
+
+	for _, endpoint := range hnsEndpoints {
+		endpointStats, err := hcsshim.GetHNSEndpointStats(endpoint.Id)
+		if err != nil {
+			log.Warnf("Failed to fetch stats for HNS endpoint %s", endpoint.Id)
+			continue
+		}
+
+		var sharedContainers = strings.Join(endpoint.SharedContainers, ",")
```

Comment:
Other metric labels contain a prefix of the runtime i.e. `containerd://...`  e.g. `windows_container_storage_read_size_bytes_total{container_id="containerd://85adcd..."} 2.569728e`

It makes sense to continue use this prefix, but the current change as is does not include a prefix and looks like
```
windows_container_network_receive_bytes_total{endpoint_id="cc1ce900-...",shared_containers="e5ce36c..."} 3.48512446e
```

Do we want to continue including the prefix for the container ids? This would mean the network stats labelling will depend on the HCS calls previously made to get the container info. If the HCS call to get container info fails then network stats labelling will also fail, and no metrics will be sent to the collector hence why in the initial change I have not included the prefix.

Shall we add the prefix based on the previous HCS calls or its fine to not include it? I can create a map to keep track of the container runtimes for the containers.

Reply:
As mentioned above, this would be a breaking change, IMO we should continue to use the `contianerd_id` and `interface` for the labels. The tooling using this information should be able to aggregate this information together in a coherent way.

Reply:
Got it, will make the suggested change as described so we don't introduce breaking changes.

Reply:
I am resolving this conversation. I have since reverted to the initial metric naming and labelling to avoid making a breaking change.